python/lsst/ts/m2/model_sim.py

```diff
@@ -29,12 +29,12 @@
 
 from lsst.ts.idl.enums import MTM2
 
-__all__ = ["Model"]
+__all__ = ["ModelSim"]
 
 
-class Model(object):
+class ModelSim(object):
```

Comment:
Since we are planning to migrate to a tcp/ip client, why not continue to have an  agnostic `Model` and move the simulator to a TCP/IP server as well? This is done on other instance like the ATDome, ATHexapod, ATSpectrograph...

Reply:
We could do this in a latter time. However, use the `ModelSim` might be an easiest way right now to support the simulation mode on Jupyter notebook. Patricio in CTIO is working on the LabVIEW TCP/IP simulator of M2 right now and I believe this might not be done in a short time (I expect this will need 2 or 3 months based on the status right now). Thanks!

Reply:
Oh, I definitely agree that we could do this later.. I just don't think we need to rename the class `ModelSim`... Just keep it as `Model` and later you updated the model to connect to the TCP/IP server... and you can move some of this code to a `Model` TCP/IP server.

Reply:
Maybe you misunderstood. The TCP/IP server (simulator) will be in LabVIEW instead of Python. I do not think it is a good idea to translate the Python code to LabVIEW code.

I could keep the name of `Model` instead of `ModelSim`. Then, I could create another class to be called as `ModelTcp` to use the `TcpClient` class in this PR. Is this what you want? I thought this might be a litter confusing but it works to me. Thanks!

Reply:
I am not saying you should translate the LabView code to python. What I think we should do is write a simple emulator, one that has the same interface and behavior of the LabView one, and we will use that one for the simulation mode. This way we will keep the model agnostic to the controller and have a way to keep track of changes on the TCP/IP interface. You can see examples of this on the components I mentioned above.

Reply:
Sure. Created the DM-30916 to track this. Thanks!

Reply:
great! thanks! Now, with that in mind, we should probably rename `ModelSim` back to `Model`

Reply:
But this implies:
1. First implement the `ModelTcp` to use the `TcpClient` (in DM-30851)
2. After the DM-30916 is done (put the `Model` into simulator), rename the `ModelTcp` to `Model`.

Is this what you want? Thanks!

Reply:
No.. I think the class should be named simply `Model`... once you implement the `MockController` (or whatever you want to call the thing that emulates the LabView TCP/IP server) then you update `Model` to communicate with them and we will be done. The CSC will be already capable of talking to the LabView controller for free!

Reply:
About your point 2, I think it is all the more reason to do it that way. If the model really change too much from one to another it means we probably didn't defined it very well (which I don't think is the case)... What I think we will probably see is a bunch of code moving from `Model` to a new class and re-implementation of some key methods. 🤞

Reply:
I will implement the DM-30851 first, therefore, I will add a `ModelTcp` first. For the `DM-30916`, I am thinking to let Patricio do it (this is because he is the person works on the LabVIEW, I should discuss with him in a latter time. This implementation will not be very soon because Patricio and I are still discussing the interface of Json packet).

What in my mind is to use the producer and consumer design pattern to deal with the JSON packet. Therefore, the interface between `ModelTcp` and `Model` (or `ModelSim`, whatever) will be very different.

Reply:
I don't think it is a good a idea to have separate `Model` classes... The whole idea behind having a model class is to isolate this sort of behavior. I would strongly suggest you plan to covent `Model` to TCP/IP when you do  DM-30851.

Reply:
Done in the commit: 27487fa. Thanks!

Reply:
The reason to do the DM-30851 first is because of the schedule of M2. I need to have something to test the TCP/IP communication with Patricio to finalize the interface. I do not think it is a good idea that to wait another 2-3 months until Patricio finishes all the works of LabVIEW and then, test the interface (if there is something wrong, we may need to change the code again). Actually, Patricio has something of LabVIEW code already. This is why I prefer to have something first in the Python to have a interface test with him to see our design now is a good idea or not. You know, it is easier to update the code while it is still in the development than the code is totally done. Thanks!

Reply:
BTW, the put of code in `Model` to a `MockController` is in DM-30916 instead of DM-30851. Thanks!